File: Samples/Calendar/js/js/scenario2-stats.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var Globalization = Windows.Globalization;
    var Calendar = Globalization.Calendar;
    var CalendarIdentifiers = Globalization.CalendarIdentifiers;
    var ClockIdentifiers = Globalization.ClockIdentifiers;

    var outputText;

    var page = WinJS.UI.Pages.define("/html/scenario2-stats.html", {
        ready: function (element, options) {
            document.getElementById("showResults").addEventListener("click", showResults);
            outputText = document.getElementById("outputText");
        }
    });

    function reportCalendarData(calendar, calendarLabel) {
        var results = calendarLabel + ": " + calendar.getCalendarSystem() + "\n";
        results += "Months in this Year: " + calendar.numberOfMonthsInThisYear + "\n";
        results += "Days in this Month: " + calendar.numberOfDaysInThisMonth + "\n";
        results += "Hours in this Period: " + calendar.numberOfHoursInThisPeriod + "\n";
        results += "Era: " + calendar.eraAsString() + "\n";
        results += "\n";
        return results;
    }

    function showResults() {
        // This scenario uses the Windows.Globalization.Calendar class to display the calendar
        // system statistics.

        // Create Calendar objects using different constructors.
        var calendar = new Calendar();
        var japaneseCalendar = new Calendar(["ja-JP"], CalendarIdentifiers.japanese, ClockIdentifiers.twelveHour);
        var hebrewCalendar = new Calendar(["he-IL"], CalendarIdentifiers.hebrew, ClockIdentifiers.twentyFourHour);

    // Display results
    outputText.innerText =
        reportCalendarData(calendar, "User's default calendar system") +
        reportCalendarData(japaneseCalendar, "Calendar system") +
        reportCalendarData(hebrewCalendar, "Calendar system");
    }
})();

File: Samples/Calendar/cppwinrt/Scenario2_Stats.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario2_Stats.h"
#include "Scenario2_Stats.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Globalization;
using namespace Windows::UI::Xaml;

namespace
{
    void ReportCalendarData(std::wostream& results, Calendar const& calendar, PCWSTR calendarLabel)
    {
        results << calendarLabel << L": " << std::wstring_view(calendar.GetCalendarSystem()) << L"\n" <<
            L"Months in this Year: " << calendar.NumberOfMonthsInThisYear() << L"\n" <<
            L"Days in this Month: " << calendar.NumberOfDaysInThisMonth() << L"\n" <<
            L"Hours in this Period: " << calendar.NumberOfHoursInThisPeriod() << L"\n" <<
            L"Era: " << std::wstring_view(calendar.EraAsString()) << L"\n\n";
    }
}

namespace winrt::SDKTemplate::implementation
{
    Scenario2_Stats::Scenario2_Stats()
    {
        InitializeComponent();
    }

    void Scenario2_Stats::ShowResults_Click(IInspectable const&, RoutedEventArgs const&)
    {
        // This scenario uses the Windows.Globalization.Calendar class to display the parts of a date.

        // Create Calendar objects using different constructors.
        Calendar calendar;
        Calendar japaneseCalendar({ L"ja-JP" }, CalendarIdentifiers::Japanese(), ClockIdentifiers::TwelveHour());
        Calendar hebrewCalendar({ L"he-IL" }, CalendarIdentifiers::Hebrew(), ClockIdentifiers::TwentyFourHour());

        // Generate the results
        std::wostringstream results;
        ReportCalendarData(results, calendar, L"User's default calendar system");
        ReportCalendarData(results, japaneseCalendar, L"Calendar system");
        ReportCalendarData(results, hebrewCalendar, L"Calendar system");
        OutputTextBlock().Text(results.str());
    }
}
